smartapps/smartthings/logitech-harmony-connect.src/logitech-harmony-connect.groovy

```diff
@@ -658,29 +658,75 @@ def updateDevice() {
 	def data = request.JSON
 	def command = data.command
 	def arguments = data.arguments
-
+  def type = params.deviceType
 	log.debug "updateDevice, params: ${params}, request: ${data}"
 	if (!command) {
 		render status: 400, data: '{"msg": "command is required"}'
 	} else {
 		def device = allDevices.find { it.id == params.id }
-		if (device) {
-        	if (device.hasCommand("$command")) {
+    if (device) {
+        if (validateCommand(device, type, command)) {
                 if (arguments) {
                     device."$command"(*arguments)
                 } else {
                     device."$command"()
                 }
                 render status: 204, data: "{}"
-           	} else {
-				render status: 404, data: '{"msg": "Command not supported by this Device"}'
-			}
+        } else {
+          render status: 403, data: '{"msg": "Access denied. This command is not supported by current capability.")}'
+			  }
 		} else {
```

Comment:
Sorry about that. The indentation is making it confusing. However, we still have an extra curly brace? 
```
def updateDevice() {
	def data = request.JSON
	def command = data.command
	def arguments = data.arguments
 	def type = params.deviceType
	log.debug "updateDevice, params: ${params}, request: ${data}"
	
	if (!command) {
		render status: 400, data: '{"msg": "command is required"}'
	} else {
		def device = allDevices.find { it.id == params.id }
	}
    
    if (device) {
        if (validateCommand(device, type, command)) {
            if (arguments) {
                device."$command"(*arguments)
            } else {
                device."$command"()
            }
            render status: 204, data: "{}"
	        } else {
	          render status: 403, data: '{"msg": "Access denied. This command is not supported by current capability.")}'
			}
		} else {
			render status: 404, data: '{"msg": "Device not found"}'
		}
	}
}
```

The last curly brace is not needed?